datahub/dsr.py

```diff
@@ -40,24 +35,23 @@ class Config:
         allow_population_by_field_name = True
 
 
-def validate_dsr_sizes(data: dict[str, Any]) -> list[str]:  # type: ignore[misc]
+def validate_dsr_arrays(data: dict[str, Any]) -> list[str]:  # type: ignore[misc]
```

Comment:
We have `disallow_any_explicit` set to true for mypy, so I'm pretty sure it would be that. Unfortunately, the input data for that is the return type for a pydantic method, so we have to keep it.

It probably does make sense to raise an error in a function called "validate". I guess this function is better described as `get_invalid_fields`

Reply:
A recent trick I learned was to use `object` in place of `Any`. Unintuitively this is the way of telling mypy that there may be any type given and then you can use typenarrowing if you need to work with the data in a specific form. The difference is that `Any` tells mypy to completely opt-out of doing any checking for that data.

Reply:
For what I can tell, the type of `data` can only be `str | list`, so why not use that as annotation rather than `Any`?